File: lectures/lec05/namespace/namespace.py
class MathLib:
    @staticmethod
    def add(a, b):
        return a + b

    class Advanced:
        @staticmethod
        def power(base, exp):
            result = 1.0
            for _ in range(exp):
                result *= base
            return result

print("Add:", MathLib.add(5, 3))
print("Power:", MathLib.Advanced.power(2.0, 4))

File: lectures/lec05/namespace/namespace.cpp
include <iostream>

// Define a custom namespace
namespace MathLib {
    int add(int a, int b) {
        return a + b;
    }

    namespace Advanced {
        double power(double base, int exp) {
            double result = 1.0;
            for (int i = 0; i < exp; ++i) {
                result *= base;
            }
            return result;
        }
    }
}

int main() {
    // Accessing the function from MathLib
    std::cout << "Add: " << MathLib::add(5, 3) << std::endl;

    // Accessing the function from nested namespace MathLib::Advanced
    std::cout << "Power: " << MathLib::Advanced::power(2.0, 4) << std::endl;

    return 0;
}

